File: VSTemplate/PDMFrameworkAddInSample/AddIn.cs
using System;
using System.IO;
using System.Runtime.InteropServices;
using BlueByte.SOLIDWORKS.PDMProfessional.SDK;
using BlueByte.SOLIDWORKS.PDMProfessional.SDK.Attributes;
using BlueByte.SOLIDWORKS.PDMProfessional.SDK.Diagnostics;
using EPDM.Interop.epdm;
using SimpleInjector;

namespace SOLIDWORKSPDMAddIn
{

    public enum Commands
    {
        CommandOne = 1000
    }
    
    [Menu((int)Commands.CommandOne, "This is a menu item - PDMFramework")]
    [Name("Sample Addin - PDMFramework")]
    [Description("This is a description - PDMFramework")]
    [CompanyName("Blue Byte Systems Inc")]
    [AddInVersion(false, 1)]
    [IsTask(false)]
    [RequiredVersion(10, 0)]
    [ComVisible(true)]
    [Guid("721C78F4-362E-45EC-B313-CD0E33A87D67")]
    public partial class AddIn : AddInBase
    {

        public override void OnCmd(ref EdmCmd poCmd, ref EdmCmdData[] ppoData)
        {
            base.OnCmd(ref poCmd, ref ppoData);

            // todo: add implementation  
            throw new NotImplementedException("This addin has no implementation");
        }


        protected override void OnLoggerTypeChosen(LoggerType_e defaultType)
        {
            base.OnLoggerTypeChosen(LoggerType_e.File);
        }

        protected override void OnRegisterAdditionalTypes(Container container)
        {
            // register types with the container 
        }

        protected override void OnLoggerOutputSat(string defaultDirectory)
        {
            // set the logger default directory - ONLY USE IF YOU ARE NOT LOGGING TO PDM
        }
        protected override void OnLoadAdditionalAssemblies(DirectoryInfo addinDirectory)
        {
            base.OnLoadAdditionalAssemblies(addinDirectory);
        }

        protected override void OnUnhandledExceptions(bool catchAllExceptions, Action<Exception> logAction = null)
        {
            this.CatchAllUnhandledException = false;

            logAction = (Exception e) => {

                throw new NotImplementedException();
            };


            base.OnUnhandledExceptions(catchAllExceptions, logAction);
        }
    }
}

File: VSTemplate/SOLIDWORKSPDMAddInVBNET/AddIn.vb
Imports System
Imports System.IO
Imports System.Runtime.InteropServices
Imports BlueByte.SOLIDWORKS.PDMProfessional.SDK
Imports BlueByte.SOLIDWORKS.PDMProfessional.SDK.Attributes
Imports BlueByte.SOLIDWORKS.PDMProfessional.SDK.Diagnostics
Imports EPDM.Interop.epdm
Imports SimpleInjector

Namespace SOLIDWORKSPDMAddInVBNET
    Public Enum Commands
        CommandOne = 1000
    End Enum

    <Menu(CInt(Commands.CommandOne), "This is a menu item created by the SDK")>
    <Name("Sample Addin")>
    <Description("This is a description")>
    <CompanyName("Blue Byte Systems Inc")>
    <AddInVersion(False, 1)>
    <IsTask(False)>
    <RequiredVersion(10, 0)>
    <ComVisible(True)>
    <Guid("721C78F4-362E-45EC-B313-CD0E33A87D67")>
    Partial Public Class AddIn
        Inherits AddInBase

        Public Overrides Sub OnCmd(ByRef poCmd As EdmCmd, ByRef ppoData As EdmCmdData())
            MyBase.OnCmd(poCmd, ppoData)
            Throw New NotImplementedException("This addin has no implementation")
        End Sub

        Protected Overrides Sub OnLoggerTypeChosen(ByVal defaultType As LoggerType_e)
            MyBase.OnLoggerTypeChosen(LoggerType_e.File)
        End Sub

        Protected Overrides Sub OnRegisterAdditionalTypes(ByVal container As Container)
        End Sub

        Protected Overrides Sub OnLoggerOutputSat(ByVal defaultDirectory As String)
        End Sub

        Protected Overrides Sub OnLoadAdditionalAssemblies(ByVal addinDirectory As DirectoryInfo)
            MyBase.OnLoadAdditionalAssemblies(addinDirectory)
        End Sub

        Protected Overrides Sub OnUnhandledExceptions(ByVal catchAllExceptions As Boolean, ByVal Optional logAction As Action(Of Exception) = Nothing)
            Me.CatchAllUnhandledException = False
            logAction = Sub(ByVal e As Exception)
                            Throw New NotImplementedException()
                        End Sub

            MyBase.OnUnhandledExceptions(catchAllExceptions, logAction)
        End Sub
    End Class
End Namespace
